### PR TITLE
fix(permissions): manage permissions on the proxy level in webkit

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/webkit/wkBrowser.ts [hunk 1]
```diff
@@ -263,20 +263,13 @@ export class WKBrowserContext extends BrowserContextBase {
...
-    ]);
-    const filtered = permissions.map(permission => {
-      const protocolPermission = webPermissionToProtocol.get(permission);
-      if (!protocolPermission)
-        throw new Error('Unknown permission: ' + permission);
-      return protocolPermission;
-    });
-    await this._browser._browserSession.send('Playwright.grantPermissions', { origin, browserContextId: this._browserContextId, permissions: filtered });
+    for (const page of this.pages())
+      await (page._delegate as WKPage)._grantPermissions(origin, permissions);
```
Comment (yury-s) [inline src/webkit/wkBrowser.ts]: Use Pomise.all() ?

File: src/webkit/wkBrowser.ts [hunk 2]
```diff
@@ -263,20 +263,13 @@ export class WKBrowserContext extends BrowserContextBase {
...
   }
 
   async _doClearPermissions() {
-    await this._browser._browserSession.send('Playwright.resetPermissions', { browserContextId: this._browserContextId });
+    for (const page of this.pages())
+      await (page._delegate as WKPage)._clearPermissions();
```
Comment (yury-s) [inline src/webkit/wkBrowser.ts]: Use Pomise.all() ?

**Review by yury-s — APPROVED** [review]

File: src/webkit/wkBrowser.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/webkit/wkBrowser.ts]: Done.

File: src/webkit/wkBrowser.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/webkit/wkBrowser.ts]: Done.

**Pull request merged**